### PR TITLE
add(epp) : applications-nginx-plus-restapi EN

### PR DESCRIPTION
**Pull request by psamecentreon**

### CODE REVIEW

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+<!--Connections-->
+
+| Metric name                 | Description                    |
+| :-------------------------- | :----------------------------- |
+| connections.active.count    | Number of active connections   |
+| connections.idle.count      | Number of idle connections     |
+| connections.accepted.count  | Number of accepted connections |
+| connections.dropped.count   | Number of dropped connections  |
+
+
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:36]:
```suggestion
```

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+<!--Ssl-->
+
+| Metric name                      | Description                     |
+| :------------------------------- | :------------------------------ |
+| ssl.handshakes.succeeded.count   | Number of handshakes succeeded  |
+| ssl.handshakes.failed.count      | Number of handshakes failed     |
+| ssl.sessions.reuses.count        | Number of sessions reuses       |
+
+
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:56]:
```suggestion
```

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Prerequisites
+
+### API Privileges
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:61]:
```suggestion
```

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+A service account is required to request the Nginx Plus API. It needs to have sufficient reading privileges in the environment.
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:63]: Link to the official documentation ?

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+## Installation
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:65]:
```suggestion
## Setup
```

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+```
+
+2. On the Centreon Central server, install the Centreon Plugin-Pack from the RPM:
+
+```bash
+yum install centreon-pack-applications-nginx-plus-restapi.noarch
+```
+
+3. On the Centreon Web interface in "Configuration > Plugin packs > Manager", install the *Nginx Plus* Plugin-Pack
+
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:94]:
```suggestion
```

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Configuration
+
+### Host
+
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:101]:
```suggestion
```

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+This Plugin-Pack is designed to have in Centreon one host per Nginx Plus environment.
+Adding a host into Centreon, link it to the template named *App-Nginx-Plus-Restapi-custom*.
+Once the template applied, some Macros have to be configured:
+| Mandatory | Name        | Description                                      |
+| :-------- | :---------- | :----------------------------------------------- |
+| X         | APIPORT     | Port used (Default: 443)                         |
+| X         | APIPROTO    | Specify https if needed (Default: 'https')       |
+| X         | APIUSERNAME | Nginx basic username                             |
+| X         | APIPASSWORD | Nginx basic password.                            |
+| X         | APIPATH     | Specify api path (Default: '/api/6')             |
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:111]: No EXTRAOPTION available ?

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+	--filter-counters='mycountersfilter' \
+	--warning-active='60' \
+	--critical-active='80'
+	--warning-idle='8' \
+	--critical-idle='10' \
+    --warning-accepted='50' 
+	--critical-accepted='65' \
+	--warning-dropped='3' \
+    --critical-dropped='5' \
+	--verbose
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:139]: Align the options

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+| X         | APIPORT     | Port used (Default: 443)                         |
+| X         | APIPROTO    | Specify https if needed (Default: 'https')       |
+| X         | APIUSERNAME | Nginx basic username                             |
+| X         | APIPASSWORD | Nginx basic password.                            |
+| X         | APIPATH     | Specify api path (Default: '/api/6')             |
+
+
+## FAQ
+
+### How to check in the CLI that the configuration is OK and what are the main options for ?
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:116]:
```suggestion
### How to test the Plugin and what are the main options for ?
```

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md (hunk 9 above)
Comment (pkriko) [reply en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:139]: No --hostname option required ?

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+```bash
+/usr/lib/centreon/plugins/centreon_nginx_plus_restapi.pl \
+    --plugin=apps::nginx::nginxplus::restapi::plugin \
+	--mode=connections \
+	--port='443' \
+	--proto='https' \
+	--api-username='myapiuser' \
+	--api-password='myapipassword' \
+	--api-path='/api/6' \
+	--filter-counters='mycountersfilter' \
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:130]:
```suggestion
```

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+	--verbose
+	
+
+OK: Active : 5, Idle : 0, Accepted : 5, Dropped : 0|
+'connections.active.count'=5;;60;80; 'connections.idle.count'=1;;8;10; 'connections.accepted.count'=5;;50;65; 'connections.dropped.count'=0;;3;5;
+```
+
+The above command controls the Nginx Plus connections (```--mode=connections```).
+This command will trigger a WARNING alarm if the active connections increase to 60 (```--warning-active='60'```)
+and a CRITICAL alarm if it increases to 80 (```--critical-active='80'```).
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:148]: Enhance the description of the command (mode/plugin path/).

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+
+
+All the options that can be used with this plugin can be found over the ```--help``` command:
+
+```/usr/lib/centreon/plugins/centreon_nginx_plus_restapi.pl --plugin=apps::nginx::nginxplus::restapi::plugin --mode=connections --help```
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:153]:
```suggestion
```
/usr/lib/centreon/plugins/centreon_nginx_plus_restapi.pl --plugin=apps::nginx::nginxplus::restapi::plugin --mode=connections --help
```
```

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+
+
+### How do I remove the *count* perfdatas if I want to filter on just one application ?
+
+The Plugin adds the count of objects by default. This can be useless if the objects are filtered with the ```--filter-name``` parameter.
+Therefore, these useless perfdatas can be omitted by adding a perfdata filter : ```--filter-perfdata='^$'```.
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:159]: Could you add a part on the most common errors :
https://docs.centreon.com/current/en/integrations/plugin-packs/procedures/network-cisco-meraki-restapi.html#why-do-i-get-the-following-error

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+| http.$name.zone.responses.total.count                   | Number total of http-zone responses                      |
+| http.$name.zone.responses.[1xx,2xx,3xx,4xx,5xx].count   | Number 1xx,2xx,3xx,4xx,5xx of http-zone responses        |
+
+<!--Ssl-->
+
+| Metric name                      | Description                     |
+| :------------------------------- | :------------------------------ |
+| ssl.handshakes.succeeded.count   | Number of handshakes succeeded  |
+| ssl.handshakes.failed.count      | Number of handshakes failed     |
+| ssl.sessions.reuses.count        | Number of sessions reuses       |
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:54]: SSL Handshakes and SSL sessions ;-)

**Review by pkriko — CHANGES_REQUESTED** [review]

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md (hunk 4 above)
Comment (psamecentreon) [reply en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:59]: Added link of official documentation.

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md (hunk 14 above)
Comment (psamecentreon) [reply en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md]: Added informatino on most common errors.

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md (hunk 13 above)
Comment (psamecentreon) [reply en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md]: Done!

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md (hunk 8 above)
Comment (psamecentreon) [reply en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md]: Macro Extraoption added.

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md (hunk 9 above)
Comment (psamecentreon) [reply en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md]: Added --hostname option

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md (hunk 12 above)
Comment (psamecentreon) [reply en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md]: Added description of the command

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 16]
```diff
@@ -0,0 +1,169 @@
...
+	--warning-dropped='3' \
+	--critical-dropped='5' \
+	--verbose
+
+OK: Active : 5, Idle : 0, Accepted : 5, Dropped : 0|
+'connections.active.count'=5;;60;80; 'connections.idle.count'=1;;8;10; 'connections.accepted.count'=5;;50;65; 'connections.dropped.count'=0;;3;5;
+```
+
+The above command check the Nginx Plus connections (```--mode=connections```) having the Nginx basic username _myapiuser_ (```--api-username='myapiuser'```), 
+as a password _myapipassword_ (```--api-password='myapipassword'```) and API URL path by default _/api/6_ (```--api-path='/api/6'```).
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:140]:
```suggestion
The above command checks the Nginx PLus connections (```--mode=connections```).
It uses an api-username (```--api-username='myapiuser'```), an api-password (```--api-password='myapipassword'```)
and it connects to the host *mynginxplus.com* (```--hostname='mynginxplus.com'```) on the port 443 (```--port='/443'```)
on the url */api/6* (```--api-path='/api/6'```) using https (```--proto='https'```).
```

File: en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md [hunk 17]
```diff
@@ -0,0 +1,169 @@
...
+### How to test the Plugin and what are the main options for ?
+
+Once the Plugin installed, log into your poller using the *centreon-engine* user account and test 
+by running the following command (Parameters such as ```api-username``` or ```api-path``` have to be adjusted):
+
+```bash
+/usr/lib/centreon/plugins/centreon_nginx_plus_restapi.pl \
+	--plugin=apps::nginx::nginxplus::restapi::plugin \
+	--mode=connections \
+	--hostname='mynginxplus.com'
```
Comment (pkriko) [inline en/integrations/plugin-packs/procedures/applications-nginx-plus-restapi.md:119]:
```suggestion
	--hostname='mynginxplus.com' \
```

**Review by pkriko — CHANGES_REQUESTED** [review]

**Review by pkriko — APPROVED** [review]

**Pull request merged**